### DISCUSSION TITLE
Catalog Bars reading first and last record

### DISCUSSION THREAD

**Discussion by rsmb7z** [nautechsystems/nautilus_trader#828] [Q&A] [answered]
Hi,
Is there a way to get first and last bar for given bar_type from the Catalog (ordered by ts_event/ts_init)?

**Comment by SeverinJB** [comment, marked as answer]
Hi @rsmb7zhx,

I am still getting up to speed with Nautilus myself, so this might not be 100% what you are looking for. With this being said, you can get the first bar (oldest) and the last bar (newest) from the cache with the below.

```
self.cache.bar(bar_type=BarType, index=0)  # Returns the newest bar.
self.cache.bar(bar_type=BarType, index=-1)  # Returns the oldest bar.
```

Best,
Severin

**Reply by rsmb7z** [reply]
Hi @SeverinJB,
Thanks for the response, welcome and I am on it for few weeks as well. From cache it is pretty simple as its python list but I was looking for the Parquet Catalog for storing historical data [which I believe is more better to maintain than cache / redis snapshots] and starting off from the last bar without loading all data first.
So far I couldn't find the built-in way for doing this. However following workaround helped me for now using SQL with DuckDB without going into Parquet syntax details.

```python
con = duckdb.connect()
bars_dataset = ds.dataset(str(path / "data" / "bar.parquet"))
ts_event = con.execute(f"SELECT MAX(ts_event) FROM bars_dataset WHERE bar_type='{str(bar_type)}'").arrow()[0].to_pylist()[0]
```
